### YamlMime:ManagedReference
items:
- uid: System.Net.NetworkInformation.PingException
  id: PingException
  children:
  - System.Net.NetworkInformation.PingException.#ctor(System.String)
  - System.Net.NetworkInformation.PingException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Net.NetworkInformation.PingException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: PingException
  nameWithType: PingException
  fullName: System.Net.NetworkInformation.PingException
  type: Class
  summary: "Исключение, которое вызывается, когда <> </> *настроек или <> </> *настроек метод вызывает метод, приводящий к исключению."
  remarks: "<xref:System.Net.NetworkInformation.Ping>Класс этого исключения для указания, что при отправке запроса управления сообщений протокола ICMP (Internet), метод вызывается методом <xref:System.Net.NetworkInformation.Ping>класса создала необработанное исключение.</xref:System.Net.NetworkInformation.Ping> </xref:System.Net.NetworkInformation.Ping> Приложения должны проверять PingException объекта, чтобы определить проблему во внутреннем исключении.       <xref:System.Net.NetworkInformation.Ping>Класса не вызывают это исключение, если запрос проверки связи ICMP завершается неудачей из-за ошибок сети, ICMP или назначения.</xref:System.Net.NetworkInformation.Ping> Для таких ошибок <xref:System.Net.NetworkInformation.Ping>возвращает <xref:System.Net.NetworkInformation.PingReply>объект с соответствующего <xref:System.Net.NetworkInformation.IPStatus>значение, заданное в <xref:System.Net.NetworkInformation.PingReply.Status%2A>свойство.</xref:System.Net.NetworkInformation.PingReply.Status%2A> </xref:System.Net.NetworkInformation.IPStatus> </xref:System.Net.NetworkInformation.PingReply> </xref:System.Net.NetworkInformation.Ping>"
  syntax:
    content: 'public class PingException : InvalidOperationException'
  inheritance:
  - System.InvalidOperationException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Net.NetworkInformation.PingException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Net.NetworkInformation.PingException
  langs:
  - csharp
  name: PingException(String)
  nameWithType: PingException.PingException(String)
  fullName: System.Net.NetworkInformation.PingException.PingException(String)
  type: Constructor
  assemblies:
  - System
  namespace: System.Net.NetworkInformation
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Net.NetworkInformation.PingException&quot;> </xref> класса, используя указанное сообщение."
  remarks: "Этот конструктор инициализирует <xref:System.Exception.Message%2A>свойство с текстом в `message` параметр.</xref:System.Exception.Message%2A> <xref:System.Exception.InnerException%2A>Свойству `null`.</xref:System.Exception.InnerException%2A>"
  syntax:
    content: public PingException (string message);
    parameters:
    - id: message
      type: System.String
      description: "Объект <xref:System.String>, описывающая ошибку.</xref:System.String>"
  overload: System.Net.NetworkInformation.PingException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Net.NetworkInformation.PingException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Net.NetworkInformation.PingException
  langs:
  - csharp
  name: PingException(SerializationInfo,StreamingContext)
  nameWithType: PingException.PingException(SerializationInfo,StreamingContext)
  fullName: System.Net.NetworkInformation.PingException.PingException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - System
  namespace: System.Net.NetworkInformation
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Net.NetworkInformation.PingException&quot;> </xref> сериализованными данными."
  remarks: "Данный конструктор вызывается в процессе десериализации для воссоздания ранее сериализованного <xref:System.Net.NetworkInformation.PingException>объекта.</xref:System.Net.NetworkInformation.PingException>"
  syntax:
    content: protected PingException (System.Runtime.Serialization.SerializationInfo serializationInfo, System.Runtime.Serialization.StreamingContext streamingContext);
    parameters:
    - id: serializationInfo
      type: System.Runtime.Serialization.SerializationInfo
      description: "Объект, содержащий сериализованные данные объекта."
    - id: streamingContext
      type: System.Runtime.Serialization.StreamingContext
      description: "Объект <xref:System.Runtime.Serialization.StreamingContext>, указывающее контекстные сведения об источнике или назначении сериализации.</xref:System.Runtime.Serialization.StreamingContext>"
  overload: System.Net.NetworkInformation.PingException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Net.NetworkInformation.PingException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Net.NetworkInformation.PingException
  langs:
  - csharp
  name: PingException(String,Exception)
  nameWithType: PingException.PingException(String,Exception)
  fullName: System.Net.NetworkInformation.PingException.PingException(String,Exception)
  type: Constructor
  assemblies:
  - System
  namespace: System.Net.NetworkInformation
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Net.NetworkInformation.PingException&quot;> </xref> класса, используя указанное сообщение и внутреннее исключение."
  remarks: "Этот конструктор инициализирует <xref:System.Exception.Message%2A>свойство с текстом в `message` параметра и <xref:System.Exception.InnerException%2A>свойство с `innerException` параметр.</xref:System.Exception.InnerException%2A> </xref:System.Exception.Message%2A>"
  syntax:
    content: public PingException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "Объект <xref:System.String>, описывающая ошибку.</xref:System.String>"
    - id: innerException
      type: System.Exception
      description: "Исключение, которое вызвало текущее исключение."
  overload: System.Net.NetworkInformation.PingException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.InvalidOperationException
  isExternal: false
  name: System.InvalidOperationException
- uid: System.Net.NetworkInformation.PingException.#ctor(System.String)
  parent: System.Net.NetworkInformation.PingException
  isExternal: false
  name: PingException(String)
  nameWithType: PingException.PingException(String)
  fullName: System.Net.NetworkInformation.PingException.PingException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Net.NetworkInformation.PingException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Net.NetworkInformation.PingException
  isExternal: false
  name: PingException(SerializationInfo,StreamingContext)
  nameWithType: PingException.PingException(SerializationInfo,StreamingContext)
  fullName: System.Net.NetworkInformation.PingException.PingException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Net.NetworkInformation.PingException.#ctor(System.String,System.Exception)
  parent: System.Net.NetworkInformation.PingException
  isExternal: false
  name: PingException(String,Exception)
  nameWithType: PingException.PingException(String,Exception)
  fullName: System.Net.NetworkInformation.PingException.PingException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Net.NetworkInformation.PingException.#ctor*
  parent: System.Net.NetworkInformation.PingException
  isExternal: false
  name: PingException
  nameWithType: PingException.PingException
